modules/ROOT/pages/indexes-for-vector-search.adoc

```diff
@@ -387,37 +387,78 @@ For example, you cannot have one xref:indexes-vector-similarity-euclidean[Euclid
+[NOTE]
+====
+Vector indexes configured with dimensionality greater than `2048` in Neo4j 5.11 should continue to work after the limitation is applied.
+====
+|
+
+d| The validation for xref:indexes-vector-similarity-cosine[cosine similarity] verifies the vector's {l2-norm} can be represented finitely in {ieee-754} *double* precision, rather than _single_ precision.
 This can lead to certain large component vectors being incorrectly indexed, and return a similarity score of `&plusmn;0.0`.
 a| // Neo4j 5.12    pending PR
 
-| The vector index `createStatement` field from xref:indexes-for-search-performance.adoc#indexes-list-indexes[`SHOW INDEXES`] does not correctly escape single quotes in index names, labels, and property keys.
+| {link-procedures-reference}#procedure_db_index_vector_queryNodes[`db.index.vector.queryNodes`] query vector validation is incorrect with a xref:indexes-vector-similarity-cosine[cosine] vector index. The {l2-norm} validation only considers the last component of the vector. If that component is `&plusmn;0.0`, an otherwise valid query vector will be thrown as invalid. This can also result in some invalid vectors being used to query, and return a similarity scores of `&plusmn;0.0`.
+
+[TIP]
+====
+For {l2-norm}alized vectors (unit vectors), thus their length image:l2norm_is_1.svg["The l2-norm of vector v equals 1"], Euclidean and cosine similarity functions produce the same similarity ordering.
+It is _recommended_ to normalize your vectors (if needed), and use a xref:indexes-vector-similarity-euclidean[Euclidean] vector index.
+====
 a| // Neo4j 5.12    pending PR
 
-a| {link-operations-manual}/backup-restore/copy-database/[Copying a database store] with a vector index does not log the recreation command, and instead logs an error:
+| The vector index `createStatement` field from xref:indexes-for-search-performance.adoc#indexes-list-indexes[`SHOW INDEXES`] does not correctly escape single quotes in index names, labels, and property keys.
+| Neo4j 5.12
+
+| {link-operations-manual}/backup-restore/copy-database/[Copying a database store] with a vector index does not log the recreation command, and instead logs an error:
 ----
 ERROR: [StoreCopy] Unable to format statement for index 'index-name'
 ----
 Due to an:
 ----
 java.lang.IllegalArgumentException: Did not recognize index type VECTOR
 ----
+
+[TIP]
+====
+If a store copy is required, please make note of the creation commands from:
+[source,cypher]
+----
+SHOW INDEXES YIELD type, createStatement
+WHERE type = 'VECTOR'
+RETURN createStatement
+----
+====
+| Neo4j 5.12
+
+| Some of the protections preventing using new features during database rolling upgrade are missing. This can result in a transaction to create a vector index being created on a cluster member running a Neo4j 5.11 version and being distributed to other cluster members on an older Neo4j version. The older Neo4j versions will fail to understand the transaction, and will panic.
```

Comment:
Is "will panic" a formal way of describing a particular behaviour? We can keep it in if it is.
```suggestion
| Some of the protections preventing the use of new features during a database rolling upgrade are missing.
This can result in a transaction to create a vector index on a cluster member running Neo4j 5.11 and distributing it to other cluster members running an older Neo4j version.
The older Neo4j versions will fail to understand the transaction.
```